packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -19,6 +19,43 @@ import { Text } from '~components/Typography';
 import { opacity } from '~tokens/global';
 import type { FontFamily } from '~tokens/global';
 
+/**
+ * Pollyfill function to get around the node 18 error
+ *
+ * This function is maintained by i18nify team. Reach out to them for any change regarding this.
+ */
+const stripTrailingZerosFromParts = (
+  parts: ReturnType<typeof formatNumberByParts>,
+): ReturnType<typeof formatNumberByParts> => {
+  const decimalPart = parts.rawParts
+    .filter(({ type }) => type === 'fraction')
+    .map(({ value }) => value)
+    .join('');
+
+  const hasFraction = parts.rawParts.some(({ type }) => type === 'fraction');
+
+  if (hasFraction && /^0+$/.test(decimalPart)) {
+    delete parts.decimal;
+    delete parts.fraction;
+    parts.rawParts = parts.rawParts.filter(({ type }) => type !== 'decimal' && type !== 'fraction');
+  }
+
+  return parts;
+};
+
+/**
+ * Wrapper that uses pollyfill of i18nify team
+ */
+const pollyfilledFormatNumberByParts: typeof formatNumberByParts = (value, options) => {
```

Comment:
Also since this is a polyfill are we conditionally calling it or are we always calling it regardless. 

eg: checking for [formatToParts](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/formatToParts) existence

Reply:
I've put conditions inside this function itself. So we can always call this instead of `formatNumberByParts`. formatToParts is supported in most major browsers we target so its fine to not check that